server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -608,11 +608,18 @@ public Index getFullyQualifiedIndex() {
         return fullyQualifiedIndex;
     }
 
-    private static Map<String, RuntimeFieldType> parseRuntimeMappings(Map<String, Object> runtimeMappings, MapperService mapperService) {
+    private static Map<String, MappedFieldType> parseRuntimeMappings(Map<String, Object> runtimeMappings, MapperService mapperService) {
         if (runtimeMappings.isEmpty()) {
             return Collections.emptyMap();
         }
-        return RuntimeFieldType.parseRuntimeFields(new HashMap<>(runtimeMappings), mapperService.parserContext(), false);
+        Map<String, RuntimeFieldType> runtimeFields = RuntimeFieldType.parseRuntimeFields(new HashMap<>(runtimeMappings),
+            mapperService.parserContext(), false);
+        Map<String, MappedFieldType> runtimeFieldTypes = new HashMap<>();
+        for (RuntimeFieldType runtimeFieldType : runtimeFields.values()) {
+            MappedFieldType fieldType = runtimeFieldType.asMappedFieldType();
+            runtimeFieldTypes.put(fieldType.name(), fieldType);
+        }
+        return Collections.unmodifiableMap(runtimeFieldTypes);
```

Comment:
I was thinking that maybe we should make FieldTypeLookup public and use it in SearchExecutionContext instead of reproducing its behavour with the runtime fields defined in the search request